server/src/main/java/org/apache/cassandra/sidecar/routes/KeyspaceRingHandler.java

```diff
@@ -0,0 +1,105 @@
+package org.apache.cassandra.sidecar.routes;
+
+import java.util.Set;
+
+import com.google.common.collect.ImmutableSet;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.net.SocketAddress;
+import io.vertx.ext.auth.authorization.Authorization;
+import io.vertx.ext.web.RoutingContext;
+import org.apache.cassandra.sidecar.acl.authorization.SidecarPermissions;
+import org.apache.cassandra.sidecar.acl.authorization.VariableAwareResource;
+import org.apache.cassandra.sidecar.cluster.CassandraAdapterDelegate;
+import org.apache.cassandra.sidecar.common.server.StorageOperations;
+import org.apache.cassandra.sidecar.common.server.data.Name;
+import org.apache.cassandra.sidecar.concurrent.ExecutorPools;
+import org.apache.cassandra.sidecar.utils.CassandraInputValidator;
+import org.apache.cassandra.sidecar.utils.InstanceMetadataFetcher;
+
+import static org.apache.cassandra.sidecar.utils.HttpExceptions.cassandraServiceUnavailable;
+import static org.apache.cassandra.sidecar.utils.HttpExceptions.wrapHttpException;
+
+/**
+ * A handler that provides ring information for a specific keyspace for the Cassandra cluster
+ */
+@Singleton
+public class KeyspaceRingHandler extends AbstractHandler<Name> implements AccessProtected
+{
+    @Inject
+    public KeyspaceRingHandler(InstanceMetadataFetcher metadataFetcher,
+                               CassandraInputValidator validator,
+                               ExecutorPools executorPools)
+    {
+        super(metadataFetcher, executorPools, validator);
+    }
+
+    @Override
+    public Set<Authorization> requiredAuthorizations()
+    {
+        String resource = VariableAwareResource.DATA_WITH_KEYSPACE.resource();
+        return ImmutableSet.of(SidecarPermissions.VIEW_CLUSTER.toAuthorization(resource));
+    }
```

Comment:
The method seems to be the only reason that this class exists. Since the `requiredAuthorizations` is configured statically, maybe extending from `RingHandler` and only overriding `requiredAuthorizations` is a more viable option.

Reply:
I have already tried it, it is not possible. since input we take is different between 2 classes `Name` vs `Void`